pkg/controllers/jobset_controller.go

```diff
@@ -73,29 +78,23 @@ func (r *JobSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	}
 
 	// Get Jobs owned by JobSet.
-	ownedJobs, err := r.getChildJobs(ctx, &js, req)
+	ownedJobs, err := r.getChildJobs(ctx, &js)
 	if err != nil {
 		log.Error(err, "getting jobs owned by jobset")
 		return ctrl.Result{}, nil
 	}
 
-	// If any jobs have failed, JobSet has failed.
+	// If any jobs have failed, execute the JobSet failure policy (if any).
```

Comment:
here we should call deleteJobs on the list of jobsToDelete identified by getChildJobs